hub/api/dataset.py

```diff
@@ -59,6 +59,7 @@ def __init__(
         else:
             write_dataset_meta(self.provider, {"tensors": []})
 
+    # TODO len should consider slice
```

Comment:
@benchislett this gave me problems as I couldn't really work with a subset of dataset easily with Pytorch.

Reply:
Yeah for sure. I'm going to fix this in my next Index PR. Do you need a patch for this PR?